src/test/com/yetanalytics/datasim/xapi/profile_test.clj

```diff
@@ -0,0 +1,180 @@
+         :satisfieds :cmi5/satisfieds
+         :completed  :cmi5/completed))
+(s/def :cmi5/completed-and-passed
+  (s/alt :completed-then-passed :cmi5/completed-then-passed
+         :passed-then-completed :cmi5/passed-then-completed))
+(s/def :cmi5/maybe-completed-then-failed
+  (s/cat :maybe-completed :cmi5/maybe-completed
+         :satisfieds      :cmi5/satisfieds
+         :failed          :cmi5/failed))
+(s/def :cmi5/failed-then-maybe-completed
+  (s/cat :failed          :cmi5/failed
+         :maybe-completed :cmi5/maybe-completed))
+(s/def :cmi5/completed-and-maybe-failed
+  (s/alt :maybe-completed-then-failed :cmi5/maybe-completed-then-failed
+         :failed-then-maybe-completed :cmi5/failed-then-maybe-completed))
+
+(s/def :cmi5/waived-session
+  (s/cat :satisfieds :cmi5/satisfieds
+         :waived     :cmi5/waived
+         :satisfieds :cmi5/satisfieds))
+(s/def :cmi5/no-result-session
+  (s/cat :launched    :cmi5/launched
+         :initialized :cmi5/initialized
+         :terminated-or-abandoned
+         :cmi5/terminated-or-abandoned))
+(s/def :cmi5/completion-no-success-session
+  (s/cat :launched    :cmi5/launched
+         :initialized :cmi5/initialized
+         :completed   :cmi5/completed
+         :satisfieds  :cmi5/satisfieds
+         :terminated-or-abandoned
+         :cmi5/terminated-or-abandoned))
+(s/def :cmi5/passed-session
+  (s/cat :launched    :cmi5/launched
+         :initialized :cmi5/initialized
+         :passed      :cmi5/passed
+         :satisfieds  :cmi5/satisfieds
+         :terminated-or-abandoned
+         :cmi5/terminated-or-abandoned))
+(s/def :cmi5/completion-passed-session
+  (s/cat :launched    :cmi5/launched
+         :initialized :cmi5/initialized
+         :completed-and-passed
+         :cmi5/completed-and-passed
+         :satisfieds  :cmi5/satisfieds
+         :terminated-or-abandoned
+         :cmi5/terminated-or-abandoned))
+(s/def :cmi5/failed-session
+  (s/cat :launched    :cmi5/launched
+         :initialized :cmi5/initialized
+         :failed      :cmi5/failed
+         :terminated-or-abandoned
+         :cmi5/terminated-or-abandoned))
+(s/def :cmi5/completion-maybe-failed-session
+  (s/cat :launched    :cmi5/launched
+         :initialized :cmi5/initialized
+         :completed-and-maybe-failed
+         :cmi5/completed-and-maybe-failed
+         :satisfieds  :cmi5/satisfieds
+         :terminated-or-abandoned
+         :cmi5/terminated-or-abandoned))
+
+(s/def :cmi5/typical-session
+  (s/alt :completion-maybe-failed-session
+         :cmi5/completion-maybe-failed-session
+         :completion-passed-session
+         :cmi5/completion-passed-session
+         :failed-session
+         :cmi5/failed-session
+         :no-result-session
+         :cmi5/no-result-session
+         :passed-session
+         :cmi5/passed-session
+         :completion-no-success-session
+         :cmi5/completion-no-success-session
+         :waived-session
+         :cmi5/waived-session))
```

Comment:
You may want to try using regular `def` and symbol names for your test-specific specs in this namespace.
At least in spec 1 (dunno about 2), spec registry by keyword is _global_ and thus we try to keep it out of tests. The good news is you could just as easily write the spec on line 53 as:

    (def cmi5-launched (partial is-cmi5-id? "launched"))

and just use symbols down the line. You only actually need to `s/def` keywords for specs that will be _keys in map specs_

Reply:
Ok I can do that

Reply:
Done